pkg/controller/common.go

```diff
@@ -305,6 +308,64 @@ func (c *SharedState) deleteOperationQueue(volumeName string) {
 	lockable.Unlock()
 }
 
+func (c *SharedState) createReplicaRequestsQueue() {
+	heap.Init(&c.priorityReplicaRequestsQueue.queue)
+	heap.Init(&c.failedReplicaRequestsQueue.queue)
+}
+
+func (c *SharedState) generateReplicaAttachmentRoutines(ctx context.Context, azr azReconciler, namespace string) error {
+	klog.V(5).Infof("Begin dequeue from replica attachment queue for nodes")
+	nodes := &v1.NodeList{}
+	err := azr.getClient().List(ctx, nodes)
+	if err != nil {
+		klog.V(5).Infof("Unable to get nodes")
+		return err
+	}
+
+	for _, node := range nodes.Items {
```

Comment:
Scheduling replicas to the correct nodes needs to factor in pod taints/toleratation and affinity/anti-affinity. This logic is currently implemented by [`func (r *ReconcileReplica) manageReplicas(ctx context.Context, volumeName string) error`](https://github.com/abhisheksinghbaghel/azuredisk-csi-driver/blob/9eedc76217f699d055040f587a7e884f1141c646/pkg/controller/replica.go#L185). Ideally, process the queues and invoke this function rather than replicate the logic here. I wonder if we should factor the replica management code into `common.go` to make it easier to consume here? @sunpa93 ?

Reply:
Ideally, I think the replica creation processes in 1) pod controller 2) replica controller 3) node availability controller should be all combined into a single function in `common.go` by leveraging the `priorityReplicaRequestsQueue` instead of maintaining 3 separate replica creation work flows.